kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/Hints.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow.ui
+
+import kotlin.reflect.KClass
+
+/**
+ * Immutable, append-only map of values that are passed down the view tree
+ * via [android.view.View.showRendering] et al. Can be used by container views
+ * to give descendants information about the context in which they're drawing.
+ */
+class Hints private constructor(
+  private val map: Map<HintKey<*>, Any>
+) {
+  constructor() : this(emptyMap())
+
+  @Suppress("UNCHECKED_CAST")
+  operator fun <T : Any> get(key: HintKey<T>): T = map[key] as? T ?: key.default
```

Comment:
Is this `Any` bound actually necessary? Every such bound I've added here I've ended up removing eventually, because sometimes feature code actually wants to represent optional values.

Reply:
I think not, trying w/o.

Reply:
Oh, yes it is, because `KClass<T : Any>`